### PR TITLE
Commit that fixes the issue #11074 by upgrading pip whenever virtuale…

### LINKED ISSUE #11074: pip should be updated when the virtualenv is first created
Judging by several reports on IRC, people who have old versions of pip on their systems (OS X ships with 1.4.1, apparently) can't install mach 0.6. To solve this, we could simply update pip whenever we create the virtualenv.

### PR DESCRIPTION
**Pull request by PeterZhizhin**
I have kind of resolved the issue #11074 by adding bool variable which is set to `True` if we had created the virtualenv and `False` otherwise.

Then it updates pip by executing `pip install --upgrade pip` in the same way as packages are updated. I am a little bit worried that I have almost duplicated the installation routine from the `for` loop but I am not sure whether I should add a function or not.

I think it is the best way of doing this because it does not need any Internet access for regular work (only for the first time you execute mach) as @larsbergstrom worried [here](https://github.com/servo/servo/pull/11149). It also doesn't add any extra latency on a no-op build.

I have checked the solution inside a docker container based on debian wheezy. Before the patch `./mach` failed to run because it wasn't able to install some packages. Now it runs successfully.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #11074 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because it changes only mach_bootstrap.py

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14452)

### CODE REVIEW

**Comment by jdm** [comment]
Feel free to reassign if this isn't your jam, @frewsxcv.

File: python/mach_bootstrap.py [hunk 1]
```diff
@@ -143,6 +146,20 @@ def _activate_virtualenv(topdir):
         os.path.join("tests", "wpt", "harness", "requirements_servo.txt"),
     ]
 
+    if need_pip_upgrade:
+        # Upgrade pip when virtualenv is created to fix the issue
+        # https://github.com/servo/servo/issues/11074
+        pip = _get_exec_path(PIP_NAMES, is_valid_path=check_exec_path)
+        if not pip:
+            sys.exit("Python pip is either not installed or not found in virtualenv.")
```
Comment (frewsxcv) [inline python/mach_bootstrap.py]: At this point, we're already in a virtualenv right? Shouldn't this always have pip?

Comment (PeterZhizhin) [inline python/mach_bootstrap.py]: At this point we have pip inside virtualenv. But the problem could be that it's version is too old.
To solve the issue, we update it.

Comment (frewsxcv) [inline python/mach_bootstrap.py]: Right, that part makes sense. I'm wondering why there is this `if not pip:` check on line 153 if pip is always included in a virtualenv? Not a big deal, just curious if I'm overlooking something.

Comment (PeterZhizhin) [inline python/mach_bootstrap.py]: I don't really know. @broadcrawford added such check on Sep 18, 2015 for all package installations (you can check them in the code below). I just copy-pasted them.

**Comment by frewsxcv** [comment]
@bors-servo r+

Thanks @PeterZhizhin!

**Pull request merged**